### PR TITLE
CAR-6724 buy now changes

### PR DESCRIPTION
**Pull request by lkappeler**
References CAR-6724

## Motivation and context

- fix: return the actual response form the api
- feat: add the endpoint the update the buy now application

### CODE REVIEW

**Review by ufkaya — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -197,6 +197,7 @@ export { sendMoneybackApplication } from "./services/car/moneybackApplication"
 export {
   sendBuyNowApplication,
   fetchBuyNowConfiguration,
+  markBuyNowApplicationAsPayed,
```
Comment (talamcol) [inline src/index.ts:200]: i think the correct tense is "paid"
https://brians.wsu.edu/2016/05/30/payed-paid/

**Review by talamcol — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -249,6 +249,7 @@ Also accompanying modes and param types, as well as default values, are exported
 - [Buy Now](https://carforyou-service.preprod.carforyou.ch/swagger-ui/index.html)
   - [`sendBuyNowApplication`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Buy%20Now%20Application)
   - [`fetchBuyNowConfiguration`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Buy%20Now%20Configuration)
+  - [`markBuyNowApplicationAsPayed`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Buy%20Now%20Application/markAsPaidUsingPOST)
```
Comment (Averethel) [inline README.md:252]:
```suggestion
  - [`markBuyNowApplicationAsPaid`](https://buyer-service.preprod.carforyou.ch/swagger-ui/index.html#/Buy%20Now%20Application/markAsPaidUsingPOST)
```

File: src/index.ts (hunk 1 above)
Comment (Averethel) [inline src/index.ts:200]:
```suggestion
  markBuyNowApplicationAsPaid,
```

File: src/services/car/__tests__/buyNow.test.ts [hunk 3]
```diff
@@ -1,4 +1,8 @@
-import { fetchBuyNowConfiguration, sendBuyNowApplication } from "../buyNow"
+import {
+  fetchBuyNowConfiguration,
+  markBuyNowApplicationAsPayed,
```
Comment (Averethel) [inline src/services/car/__tests__/buyNow.test.ts:3]:
```suggestion
  markBuyNowApplicationAsPaid,
```

File: src/services/car/__tests__/buyNow.test.ts [hunk 4]
```diff
@@ -99,4 +105,14 @@ describe("Car API", () => {
       expect(fetch).toHaveBeenCalled()
     })
   })
+
+  describe("markBuyNowApplicationAsPayed", () => {
```
Comment (Averethel) [inline src/services/car/__tests__/buyNow.test.ts:109]:
```suggestion
  describe("markBuyNowApplicationAsPaid", () => {
```

File: src/services/car/buyNow.ts [hunk 5]
```diff
@@ -26,20 +30,34 @@ export const sendBuyNowApplication = async ({
...
-      result: buyNowApplication,
+      result: response,
     }
   } catch (error) {
-    return ignoreServerSideErrors({ error, returnValue: buyNowApplication })
+    return handleValidationError(error)
   }
 }
 
+export const markBuyNowApplicationAsPayed = async ({
```
Comment (Averethel) [inline src/services/car/buyNow.ts:47]:
```suggestion
export const markBuyNowApplicationAsPaid = async ({
```

**Review by GutierrezGo — APPROVED** [review]

**Review by Averethel — APPROVED** [review]
Looks good to me except for the wrong past tense 

Not sure if you know and use it, but [code spell checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) had been a life saver to me when it comes to typos in variable names.

**Comment by lkappeler** [comment]
> Not sure if you know and use it, but code spell checker had been a life saver to me when it comes to typos in variable names.

Yeah I have that but it did not lint it for some reason... (Had in mind pay conjugates irregular such as display..) thanks for the hint 🙂

**Pull request merged**